src/main/java/org/nypl/findawayclientlibrary/PlayBookActivity.java

```diff
@@ -359,15 +260,27 @@ protected void onPause() {
   protected void onResume() {
     super.onResume();
 
-
     // a stream of _all_ download events for the supplied content id
     // the onCompleted(), onError() and onNext() methods are the ones implemented in the activity itself.
-    eventsSubscription = downloadService.subscribeDownloadEventsAll(downloadService, contentId);
+    downloadService.subscribeDownloadEventsAll(downloadService, contentId);
+
+    // We know what book this activity is to be playing.  Does this book need any downloading?
+    // Check to see all the book files have successfully downloaded.
+    Integer downloadStatus = downloadService.getDownloadStatus(contentId);
+    // If not, ask the DownloadService to either start or resume the download.
+    if (downloadStatus.equals(DownloadService.DOWNLOAD_ERROR) || downloadStatus.equals(DownloadService.DOWNLOAD_NEEDED)) {
+      // ask to start the download
+      downloadService.downloadAudio(contentId, license, currentlyDownloadingChapter, null);
```

Comment:
Will the `PlaybackActivity` always retain these values directly? this would be another place for it to use the model, that way we would change all of these values at once. 

also does it matter if the `part` is `null`?  I don't see anything in their documentation regarding this method signature but that could lead to undefined behavior.

Reply:
Yes, the values are hardcoded now, but should be coming from the manifest via the model.

Findaway expects the part to be a valid positive int, so in the downloadAudio method I check the null, and substitute a default integer value for it.  It's a way for me to flag that I don't know what part I stopped on, and am ok with passing in whatever.  I could write code to track chapter parts more closely, but I don't feel I need to -- the sdk will skip already-downloaded audio for me.